contracts/truefi2/SAFU.sol

```diff
@@ -25,6 +25,8 @@ contract SAFU is UpgradeableClaimable {
         require(loanFactory.isLoanToken(address(loan)), "SAFU: Unknown loan");
         require(loan.status() == ILoanToken2.Status.Defaulted, "SAFU: Loan is not defaulted");
         ITrueFiPool2 pool = ITrueFiPool2(loan.pool());
-        IERC20(pool.token()).safeTransfer(address(pool), loan.debt());
+        pool.liquidate(loan);
+        uint256 lostByPool = loan.debt().mul(loan.balanceOf(address(this))).div(loan.totalSupply());
+        IERC20(pool.token()).safeTransfer(address(pool), lostByPool);
```

Comment:
`pool.liquidate` will transfer loan tokens to the SAFU contract. Maybe we could approve pool by SAFU for `lostByPool` amount of token, then `pool.liquidate` would make a `transferFrom` and if it passes the pool would transfer loan tokens to SAFU. This way pools will be a little more distrustful towards already centralized SAFU.

Reply:
not a bad idea, but we would still need to pass compensation amount to the pool. What if it's 0? This change will not give us any additional protection, i think